morphia/src/main/java/org/mongodb/morphia/converters/BigDecimalConverter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.morphia.converters;
+
+import org.bson.types.Decimal128;
+import org.mongodb.morphia.mapping.MappedField;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * This provides a convert to/from BigDecimal in applications using the Decimal128 type introduced in the 3.4.0 release of MongoDB.
```

Comment:
This doesn't jibe with the actual implementation, which decodes a whole bunch of types.  Question about that below.